python/dgl/graph.py

```diff
@@ -1101,6 +1101,64 @@ def add_edges(self, u, v, data=None):
             self._msg_index = self._msg_index.append_zeros(num)
         self._msg_frame.add_rows(num)
 
+    def del_nodes(self, vids):
+        """Remove multiple nodes.
+
+        Parameters
+        ----------
+        vids: list, tensor
+            The id of nodes to remove.
+        """
+        if self.is_readonly:
+            raise DGLError("del_nodes is not supported by read-only graph.")
+        induced_nodes = utils.set_diff(utils.toindex(self.nodes()), utils.toindex(vids))
+        sgi = self._graph.node_subgraph(induced_nodes)
+
+        if isinstance(self._node_frame, FrameRef):
+            self._node_frame = FrameRef(Frame(self._node_frame[sgi.induced_nodes]))
+        else:
+            self._node_frame = FrameRef(self._node_frame, sgi.induced_nodes)
+
+        if isinstance(self._edge_frame, FrameRef):
+            self._edge_frame = FrameRef(Frame(self._edge_frame[sgi.induced_edges]))
+        else:
+            self._edge_frame = FrameRef(self._edge_frame, sgi.induced_edges)
+
+        # FIXME(zihao): the following impl is not efficient, but directly reusing the handle of subgraph index would
```

Comment:
@jermainewang and @zheng-da , could you please have a look at this? After discussion with @BarclayII we think the problem resides in "GraphInterface.Reset".